sway-core/src/type_engine/engine.rs

```diff
@@ -47,6 +47,60 @@ impl Engine {
             (Ref(received), _) => self.unify(received, expected, span),
             (_, Ref(expected)) => self.unify(received, expected, span),
 
+            (Unknown, Numeric) => match self.slab.replace(
```

Comment:
why does this resolve with `u64` and not `Numeric`?

Reply:
Fixed.